### ISSUE TITLE
Drawing Primitives

### ISSUE THREAD

**Issue by henry-paschke** [MarkelZ/pygame-render#8]
Is there any way to draw primitives such as lines or polygons? I don't see anything in the docs or examples on how to do this.

**Comment by MarkelZ** [comment]
Hey @krakengamedev! The engine's main purpose is to render textures on the screen using shaders. To render a primitive, you can create a texture with the shape of that primitive and render it with the engine. 

I just created a small example where I generate the texture of a white circle over a black background using `pygame`, and then I render this texture with a shader that replaces black pixels with transparent ones and the white ones with whichever color you want. Check this out: [krakengamedev.zip](https://github.com/MarkelZ/pygame-render/files/15138535/krakengamedev.zip). You can use this method for polygons too. Instead of generating the textures with `pygame` you can also have these shapes stored as `.png` files and load them with `load_texture`. 

Lines may be a bit harder to draw. You can do it by scaling and rotating a rectangle since `render` lets you apply transformations, but feels a bit hacky. I'll look into implementing primitives explicitly, I'll keep you posted. Thank you for your feedback :)

**Comment by henry-paschke** [comment]
That makes sense, I was thinking the same thing myself with rendering the lines ahead of time and then applying transforms, but that's really not ideal. 
With the modernGL backend I would assume that primitives would be possible, but I'm not sure. Either way, thanks and keep me updated!

**Comment by MarkelZ** [comment]
Hi again @henry-paschke! I just uploaded version 1.3.0 to PyPI, you can install it with `pip install pygame-render==1.3.0`. It has functionality for drawing circles, triangles, rectangles, circle arcs, lines, and thick lines. Check out [example8.py](https://github.com/MarkelZ/pygame-render/blob/main/examples/example8.py) to see a demo. Hope it's useful!

**Comment by henry-paschke** [comment]
Awesome, thank you for the continued support! I'm looking forward to integrating this into my engine.

**Comment by MarkelZ** [comment]
It sounds interesting, what kind of engine is it? Is the source code public?

**Comment by henry-paschke** [comment]
It is a python + C++ engine for procedurally animated and rigged characters. The source code is private for now unfortunately. I started using this library for rendering initially, but most of the math I needed to do was too slow in python, and the physics engine I was using, Pymunk, is natively in C. So I rewrote the entire thing in C++, using SFML for my rendering. But all my animations are still made with my pygame animation editor. I'm using your library for that editor because it still utilizes all the original python code, its so much faster to write demo code/editing tools in python, but I need blazing fast rendering.

Here is a link to a demo: https://www.dropbox.com/scl/fi/fxics3uxlglqucu99u273/Release.zip?rlkey=ai1m5gzjok2k93hv9xrrxt5fb&st=ajofvl7t&dl=0 

WASD is move and jump, space bar to spawn blocks, and arrow keys to move camera.
